### PR TITLE
Support MRC and TIFF loading in `Image`

### PR DESCRIPTION
**Pull request by garrettwrong**
Still need to add tests etc.

### CODE REVIEW

**Comment by garrettwrong** [comment]
Thinking I should probably just implement this as part of `Image.load` and try to use `Image` in `apple`.

**Comment by garrettwrong** [comment]
Closes #971 #786 

786's `testFileCorruption` is now handled by a much smaller and direct test of `Image`.  This was the worst offender.

**Comment by garrettwrong** [comment]
Restarting one of these tests to see if it shares the MRC warning issue found in `develop`.

File: src/aspire/image/image.py [hunk 1]
```diff
@@ -343,6 +402,37 @@ def save(self, mrcs_filepath, overwrite=False):
...
+    @staticmethod
+    def load(filepath, dtype=None):
+        """
+        Load raw data from supported files.
+
+        Currently MRC and TIFF are supported.
+
+        :param dtype: Optionally force cast to `dtype`.
+             Default dtype is inferred from the file contents.
+        :return: numpy array of image data.
```
Comment (j-c-c) [inline src/aspire/image/image.py]: Missing `param` here.

File: src/aspire/image/image.py [hunk 2]
```diff
@@ -343,6 +402,37 @@ def save(self, mrcs_filepath, overwrite=False):
...
+        """
+
+        # Get the file extension
+        ext = os.path.splitext(filepath)[1]
+
+        # On unsupported extension, raise with suggested file types
+        if ext not in Image.extensions:
+            raise RuntimeError(
```
Comment (j-c-c) [inline src/aspire/image/image.py:423]: Should we test this branch?

File: tests/test_image.py [hunk 3]
```diff
@@ -305,3 +308,90 @@ def test_asnumpy_readonly():
     # Attempt assignment
     with raises(ValueError, match=r".*destination is read-only.*"):
         vw[0, 0, 0] = 123
+
+
+@pytest.mark.xfail(reason="Ray logging issue ray#37711", strict=False)
```
Comment (j-c-c) [inline tests/test_image.py:312]: Nice find on that one!

File: src/aspire/image/image.py [hunk 4]
```diff
@@ -343,6 +402,37 @@ def save(self, mrcs_filepath, overwrite=False):
...
+                f"Attempting to open unsupported file extension '{ext}', try {list(Image.extensions.keys())}."
+            )
+
+        # Call the appropriate file reader
+        im = Image.extensions[ext](filepath)
+
+        # Attempt casting if user
```
Comment (j-c-c) [inline src/aspire/image/image.py]: Unfinished comment?

**Review by j-c-c — CHANGES_REQUESTED** [review]
Looks good! Just a couple small things.

File: tests/test_image.py (hunk 3 above)
Comment (garrettwrong) [reply tests/test_image.py:312]: grrrr

File: src/aspire/image/image.py (hunk 4 above)
Comment (garrettwrong) [reply src/aspire/image/image.py]: yep, updated.

File: src/aspire/image/image.py (hunk 2 above)
Comment (garrettwrong) [reply src/aspire/image/image.py:423]: Sure. Added.

File: src/aspire/image/image.py (hunk 1 above)
Comment (garrettwrong) [reply src/aspire/image/image.py]: added, thanks

**Review by j-c-c — APPROVED** [review]
LGTM!

**Comment by garrettwrong** [comment]
When the CI passes, I'm planning to merge this so we can get the patch for the broken CI test in ASAP.

**Pull request merged**